smart-contract-extensions/contracts/KeyPurchaserFactory.sol

```diff
@@ -0,0 +1,56 @@
+pragma solidity 0.5.16;
+
+import './KeyPurchaser.sol';
+import 'hardlydifficult-ethereum-contracts/contracts/proxies/CloneFactory.sol';
+import 'unlock-abi-1-3/IPublicLockV6.sol';
+
+/**
+ * @notice A factory for creating keyPurchasers.
+ * This contract acts as a registry to discover purchasers for a lock
+ * and by creating each purchaser itself, it's a single tx and this guarantees
+ * a consistent implementation and that it was created by the lock owner.
+ */
+contract KeyPurchaserFactory
+{
+  using CloneFactory for address;
+
+  event KeyPurchaserCreated(address indexed forLock, address indexed keyPurchaser);
+
+  address public keyPurchaserTemplate;
+  mapping(address => address[]) internal lockToKeyPurchasers;
+
+  constructor() public
+  {
+    keyPurchaserTemplate = address(new KeyPurchaser());
+  }
+
+  /**
+   * @notice Deploys a new KeyPurchaser for a lock and stores the address for reference.
+   */
+  function deployKeyPurchaser(
+    IPublicLock _lock,
```

Comment:
shouldn't this be `IPublicLockV6` as that's what you imported?

Reply:
The name of the contract != the name of the file. We might want to fix that and republish.  I'd vote removing V6 from the filename.

Reply:
Haha, right you are. I'd second that vote!

Reply:
hmm second thought.  an advantage to renaming the contract name in the interface file is an extension could then support multiple lock versions.  without that - truffle breaks on a dupe name.